### PR TITLE
[SMTChecker] Decrease nondeterminism

### PR DESCRIPTION
**Pull request by leonardoalt**
Needed by https://github.com/ethereum/solidity/pull/10205

### CODE REVIEW

File: libsolidity/formal/SMTEncoder.cpp [hunk 1]
```diff
@@ -1804,23 +1804,25 @@ pair<smtutil::Expression, smtutil::Expression> SMTEncoder::divModWithSlacks(
...
+	smt::SymbolicIntVariable dSymb(intType, intType, "d_" + suffix, m_context);
+	smt::SymbolicIntVariable rSymb(intType, intType, "r_" + suffix, m_context);
+	auto d = dSymb.currentValue();
+	auto r = rSymb.currentValue();
 
 	// x / y = d and x % y = r iff d * y + r = x and
 	// either x >= 0 and 0 <= r < abs(y) (or just 0 <= r < y for unsigned)
 	// or     x < 0 and -abs(y) < r <= 0
-	m_context.addAssertion(((d.currentValue() * _right) + r.currentValue()) == _left);
+	m_context.addAssertion(((d * _right) + r) == _left);
```
Comment (axic) [inline libsolidity/formal/SMTEncoder.cpp:1815]: So as per my understanding `.currentValue()` generates a statement for the query (on first invocation), of which the order we want to be deterministic, but the ordering of these expression we don't care about.

**Review by axic — APPROVED** [review]
Would have been nicer if some tests (which contain the query dump) would have been affected by this, so we can confirm.

**Comment by leonardoalt** [comment]
@axic You can see that currently in the second commit of https://github.com/ethereum/solidity/pull/10205 which finally makes the tests there pass. This PR just extracted that commit.

**Comment by axic** [comment]
> You can see that currently in the second commit of #10205 which finally makes the tests there pass. This PR just extracted that commit.

Because there are no tests yet affected by this, lets merge, don't really see any downside in this one.

**Pull request merged**